.github/scripts/install.sh

```diff
@@ -48,41 +36,19 @@ else
 
     # To check our manifest and coverage
     pip install -q check-manifest codecov
-
-    # Remove packages we have subrepos for
-    for dep in $(ls external-deps)
-    do
-        echo "Removing $dep package"
-        pip uninstall $dep -q -y
-    done
-
-    # Remove Spyder to properly install it below
-    pip uninstall spyder -q -y
 fi
 
-# Install subrepos in development mode
-for dep in $(ls external-deps)
-do
-    echo "Installing $dep subrepo"
-
-    # This is necessary to pass our minimal required version of PyLSP to setuptools-scm
-    if [ "$dep" = "python-lsp-server" ]; then
-        SETUPTOOLS_SCM_PRETEND_VERSION=`python pylsp_utils.py` pip install --no-deps -q -e external-deps/$dep
-    else
-        pip install --no-deps -q -e external-deps/$dep
-    fi
-done
+# Install subrepos from source
+python -bb -X dev -W error install_dev_repos.py --not-editable --no-install spyder
 
 # Install boilerplate plugin
 pushd spyder/app/tests/spyder-boilerplate
 pip install --no-deps -q -e .
 popd
 
 # Install Spyder to test it as if it was properly installed.
-# Note: `python setup.py egg_info` doesn't work here but it
-# does locally.
-python setup.py -q bdist_wheel
-pip install --no-deps -q dist/spyder*.whl
+python -bb -X dev -W error -m build
+python -bb -X dev -W error -m pip install --no-deps dist/spyder*.whl
```

Comment:
@CAM-Gerlach, is it necessary to build the whl in a separate step? When I install spyder via `python -m pip install --no-deps .` it indicates that it builds the whl into a temporary folder which does not persist after the install. Isn't pip installing from the built whl, otherwise why build the whl in the first place just to destroy it?

```
Processing /Users/rclary/Documents/Python/spyder
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: spyder
  Building wheel for spyder (setup.py) ... done
  Created wheel for spyder: filename=spyder-5.4.0.dev0-py3-none-any.whl size=14850303 sha256=9e9f82f095f0c8b2896a87b0464d96f711a311133e08d759bc257f99d0de51b4
  Stored in directory: /private/var/folders/5v/28jqvwxs2cd5fj93gvwnykdrqc926z/T/pip-ephem-wheel-cache-zggh12lw/wheels/bb/14/5d/a67d0e97ac041166e5d8d5fa834e063ab0a12d72c322d8a11b
Successfully built spyder
Installing collected packages: spyder
  Attempting uninstall: spyder
    Found existing installation: spyder 5.4.0.dev0
    Uninstalling spyder-5.4.0.dev0:
      Successfully uninstalled spyder-5.4.0.dev0
Successfully installed spyder-5.4.0.dev0
```

Reply:
A few things: 

* It ensures the wheel is built in modern PEP 517 mode with build isolation rather than the deprecated legacy mode (though that can also be ensured with an extra flag in pip)
* It builds both the sdist and the wheel and ensures both can be built, and additionally ensures the wheel can successfully be built from the sdist rather than just directly from the source tree
* There are extra developer-focused checks and more detailed output, useful for debugging especially on CI
* It tests the wheel as built following the same process and tooling as it would be for an actual release, and ensures a clean separation of the build and install steps (regardless of pip version, options, package config, etc).

Since the same steps are accomplished, just by different frontends, it should result in any meaningful increase in build time (at least more than a second or two).